README-docker.md

```diff
@@ -0,0 +1,343 @@
+- Exploring the data
+- Manual processing steps
+
+## Environment Variables
+
+Configure the Docker wrapper using environment variables:
+
+### `MUSSEL_DOCKER_IMAGE`
+Docker image name to use
+- **Default:** `mussel:latest`
+- **Example:** `MUSSEL_DOCKER_IMAGE=mussel:v1.0 ./mussel-docker tessellate --help`
+
+### `MUSSEL_BACKEND`
+Backend to use for building the image
+- **Options:** `torch-gpu`, `torch-cpu`, `tensorflow-gpu`, `tensorflow-cpu`
+- **Default:** `torch-gpu`
+- **Example:** `MUSSEL_BACKEND=torch-cpu ./mussel-docker build`
+
+### `MUSSEL_USE_GPU`
+Enable GPU support when running commands
+- **Options:** `true`, `false`
+- **Default:** `true`
+- **Example:** `MUSSEL_USE_GPU=false ./mussel-docker extract_features ...`
+
+### `MUSSEL_WORK_DIR`
+Working directory to mount into the container
+- **Default:** Current directory (`$(pwd)`)
+- **Example:** `MUSSEL_WORK_DIR=/path/to/data ./mussel-docker tessellate ...`
+
+## Working with Files
+
+The wrapper script automatically mounts your current working directory into the container at `/data`. This means:
+
+- All file paths in commands should be relative to your current directory
+- Output files will be written to your current directory with your user's ownership
+- You can access any files in subdirectories
+
+**Important:** The Docker container runs with your user ID and group ID, ensuring that all files created by the container have the correct ownership on the host system. This prevents permission issues when accessing output files.
+
+### Example Workflow
+
+```bash
+# Create a project directory
+mkdir my-project
+cd my-project
+
+# Put your slide file here
+cp /path/to/slide.svs .
+
+# Run commands (they will access files in the current directory)
+../mussel-docker tessellate slide_path=slide.svs output_h5_path=tiles.h5
+../mussel-docker extract_features \
+    slide_path=slide.svs \
+    patch_h5_path=tiles.h5 \
+    output_h5_path=features.h5
+
+# All output files are in the current directory
+ls -lh
+```
+
+## Advanced Usage
+
+### Custom Docker Image Name
+
+Build with a custom image name:
+```bash
+MUSSEL_DOCKER_IMAGE=my-mussel:custom MUSSEL_BACKEND=torch-gpu ./mussel-docker build
+```
+
+Use the custom image:
+```bash
+MUSSEL_DOCKER_IMAGE=my-mussel:custom ./mussel-docker tessellate --help
+```
+
+### Mount Multiple Directories
+
+If you need to access files from different directories, use the interactive shell:
+
+```bash
+docker run --rm -it \
+    --user $(id -u):$(id -g) \
+    -v /path/to/slides:/slides \
+    -v /path/to/output:/output \
+    --gpus all \
+    mussel:latest \
+    /bin/bash
+```
+
+Then inside the container:
+```bash
+uv run tessellate slide_path=/slides/slide.svs output_h5_path=/output/tiles.h5
+```
+
+**Note:** The `--user $(id -u):$(id -g)` flag ensures files are created with your user's ownership.
+
+### GPU Configuration
+
+Check if GPUs are available:
+```bash
+docker run --rm --gpus all nvidia/cuda:12.1.1-cudnn8-devel-ubuntu22.04 nvidia-smi
```

Comment:
This is specifically for CUDA 12.1.1, which is what we have on the mskhpc{1,2,3} servers.   We have CUDA 12.3 on the hpcgpu{1,2,3,4} servers.  The CUDA version probably needs to match the hardware that the user intends to run on.  We should probably warn people about that.  This probably applies to the Dockerfile too.